src/coreclr/vm/prestub.cpp

```diff
@@ -1302,10 +1310,7 @@ bool PrepareCodeConfig::FinalizeOptimizationTierForTier0LoadOrJit()
         // native code version of a method. The current optimization tier should be consistent with the change being made
         // (Tier 0 to Optimized), such that the tier is not changed in an unexpected way or at an unexpected time. Since changes
         // to the optimization tier are unlocked, this assertion is just a speculative check on possible values.
-        NativeCodeVersion::OptimizationTier previousOptimizationTier = GetCodeVersion().GetOptimizationTier();
-        _ASSERTE(
-            previousOptimizationTier == NativeCodeVersion::OptimizationTier0 ||
-            previousOptimizationTier == NativeCodeVersion::OptimizationTierOptimized);
+        _ASSERTE(!GetCodeVersion().IsFinalTier());
```

Comment:
I think the previous assertion is more correct. The optimization tier could already be `OptimizationTierOptimized` due to the race mentioned in the comment above. The optimization tier here may be any initial tier or `OptimizationTierOptimized`, so after this change it may also be `OptimizationTierInstrumented`.